src/main/proto/ga4gh/bio_metadata_service.proto

```diff
@@ -0,0 +1,97 @@
+syntax = "proto3";
+
+package ga4gh;
+
+import "ga4gh/bio_metadata.proto";
+
+
+service BioMetadataService {
+  // Gets a list of Individuals accessible through the API.
+  // `POST /individuals/search` must accept a JSON version of
+  // `SearchIndividualsRequest` as the post body and will return a JSON version
+  // of `SearchIndividualsResponse`.
+  rpc SearchIndividuals(SearchIndividualsRequest)
+      returns (SearchIndividualsResponse);
+  
+  // `POST /biosamples/search` must accept a JSON version of
+  // `SearchBioSamplesRequest` as the post body and will return a JSON version
+  // of `SearchBioSamplesResponse`.
+  rpc SearchBioSamples(SearchBioSamplesRequest)
+      returns (SearchBioSamplesResponse);
+
+  // Gets an `Individual` by ID.
+  // `GET /individuals/{id}` will return a JSON version of `Individual`.
+  rpc GetIndividual(SearchIndividualsRequest)
+      returns (Individual);
+
+  // Gets a `BioSample` by ID.
+  // `GET /biosamples/{id}` will return a JSON version of `BioSample`.
+  rpc GetBioSample(SearchIndividualsRequest)
+      returns (BioSample);
+}
+// *********************  /individuals ********************************
+// This request maps to the body of `POST /individuals/search` as JSON.
+message SearchIndividualsRequest {
+  // The dataset to search within.
+  string dataset_id = 1;
+
+  // Returns Individuals with the given :ref:`name <apidesign_object_names>`
+  // found by case-sensitive string matching.
+  string name = 2;
+
+  // Specifies the maximum number of results to return in a single page.
+  // If unspecified, a system default will be used.
+  int32 page_size = 3;
+
+  // The continuation token, which is used to page through large result sets.
+  // To get the next page of results, set this parameter to the value of
+  // `nextPageToken` from the previous response.
+  string page_token = 4;
+}
+
+//This is the response from `POST /individuals/search` expressed as JSON.
+message SearchIndividualsResponse {
+  // The list of individuals.
+  repeated Individual individuals = 1;
+
+  // The continuation token, which is used to page through large result sets.
+  // Provide this value in a subsequent request to return the next page of
+  // results. This field will be empty if there aren't any additional results.
+  string next_page_token = 2;
+}
+
+// *********************  /biosamples ********************************
+
+message SearchBioSamplesRequest {
+  // The dataset to search within.
+  string dataset_id = 1;
+  
+  // Returns BioSamples with the given :ref:`name <apidesign_object_names>`
+  // found by case-sensitive string matching.
+  string name = 2;
+  
+  // Returns BioSamples for the provided individual ID.
+  string individual_id = 3;
+  
```

Comment:
> Can we add a search by disease also? This would be very useful. I'm not sure what the semantics of searching by ontology term should be though. Would exact matching do, or do we need to be more sophisticated?

I think doing this lexically would be a bad idea. Would it be possible to use an ontology service (either part of GA4GH or distinct) to first retrieve a disease identifier. The sample query could then take that identifier (rather than a label) and implement the appropriate search. A lot more needs to be specified here. What should happen in the case where we have equivalence axioms between identifiers in different disease resources (equivalent diseases can have different IDs in OMIM, Orphanet, NCIT, DO, MESH, ICD9/10/11, SNOMED, ...). What about when a generic disease that subsumes various specific diseases? If a user requests "parkinson disease" samples, then any sample annotate with any of the different forms of PD should presumably be returned, but how this should be implemented is under discussion in a different ticket.

I'm not sure what to recommend as I'm not sure what the general design pattern is for the FooRequest objects. We can imagine a fairly open-ended set of requirements for different ways to search samples. It looks like the goal is not to allow the expressivity of a full-blown query language. Instead common types of request are supported by shadowing the domain objects, allowing the client to plug in values that are interpreted as a conjunctive query.

To be honest I'm a bit confused by:

https://github.com/ga4gh/schemas/blob/2efea413027a820a53f313efa2ed5cf8ee380d31/src/main/proto/ga4gh/bio_metadata_service.proto#L64-L82

The semantics of this aren't clear.

The comment says the dataset id is optional, but it doesn't have the `optional` proto tag.

The name field is not marked optional, the docs say string matching is to be used. We might assume that an empty string would match everything (more docs are required on the type of string matching. Any substring? Lucene type queries?).

I find the choice of `name` as the field confusing here. Why not something like `nameMatch`?

The compositionality of these two fields aren't clear. I'd guess these would be conjunctive, e.g. I can search for samples matching a string AND that are in a particular dataset.

This could rapidly become confusing as the ability to do more expressive search is added (unless the goal is to deliberately restrict to highly unexpressive searches)